tssc/utils/maven.py

```diff
@@ -0,0 +1,128 @@
+"""
+Shared utils for maven deeds
+"""
+
+import xml.etree.ElementTree as ET
+
+def generate_maven_settings(working_dir, maven_servers, maven_repositories, maven_mirrors):
+
+    """
+    Generates and returns a settings.xml file from the inputs it is provided.
+
+    Parameters
+    ----------
+    :
+    maven_servers:
+        Dictionary of id, username, password
+    maven_repositories:
+        Dictionary of id, url, snapshots, releases
+    maven_mirrors:
+        Dictionary of id, url, mirror_of
+
+    Returns
+    -------
+    dict
+        Dictionary parsed from given YAML or JSON file
+
+    Raises
+    ------
+    ValueError
+        If the given file can not be parsed as YAML or JSON.
+
+    Returns
+    -------
+    Str
+        The path to the settings.xml that was generated.
+    """
+
+    root = ET.Element('settings')
+    generate_maven_servers(root, maven_servers)
+    generate_maven_repositories(root, maven_repositories)
+    generate_maven_mirrors(root, maven_mirrors)
+
+    tree = ET.ElementTree(root)
+
+    with open(working_dir + '/settings.xml', 'wb') as files:
```

Comment:
nit: `file` vs `files` since it's singular here

Reply:
done